model/src/main/kotlin/config/OrtConfiguration.kt

```diff
@@ -41,21 +44,42 @@ data class OrtConfiguration(
          * 1. [Command line arguments][args]
          * 2. [Configuration file][configFile]
          * 3. default.conf from resources
+         *
+         * The configuration file is optional and does not have to exist. However, if it exists, but does not
+         * contain a valid configuration, an [IllegalArgumentException] is thrown.
          */
-        fun load(args: Map<String, String> = emptyMap(), configFile: File? = null): OrtConfiguration {
-            val argsConfig = ConfigFactory.parseMap(args, "Command line").withOnlyPath("ort")
-            val fileConfig = configFile?.let {
-                ConfigFactory.parseFile(it).withOnlyPath("ort")
+        fun load(args: Map<String, String> = emptyMap(), configFile: File): OrtConfiguration {
+            val result = ConfigLoader.Builder()
+                .addSource(argumentsSource(args))
+                .addSource(PropertySource.file(configFile, optional = true))
+                .build()
+                .loadConfig<OrtConfigurationWrapper>()
+            return result.map { it.ort }.getOrElse {
+                if (configFile.exists()) {
```

Comment:
Error handling is indeed tricky with the API provided by the configuration library. Loading of the configuration can only fail if the OrtConfiguration instance cannot be constructed. This happens for two reasons:
* The structure of the configuration file is invalid, missing the required top-level path 'ort'
* Storages are incorrectly configured, so that mandatory properties are missing, or the storage type cannot be figured out by the properties provided.

Additional, unrecognized properties are simply ignored.

The main purpose of the map with arguments is probably to override or extend properties existing in the configuration file. When used that way it is unlikely that properties contained in the map can cause a configuration error. It would, however, be possible that a complete storage was configured (incorrectly) by command line properties. To detect this, I now check whether the map contains properties starting with the base path 'ort.*' and also throw an exception then.